application-task-ui/src/main/resources/TaskManager/EditorContentUpdater.xml

```diff
@@ -0,0 +1,543 @@
+    let currentHTMLContent = CKEDITOR.plugins.xwikiSource.getFullData(editor);
+    let convertURL = XWiki.currentDocument.getURL('get', $.param({
+      sheet: 'CKEditor.HTMLConverter',
+      outputSyntax: 'plain',
+      language: $('html').attr('lang') || '',
+      formToken: document.documentElement.dataset.xwikiFormToken || ''
+    }));
+    $.ajax({
+      url: convertURL,
+      method: 'POST',
+      data: {
+        sourceSyntax: 'xwiki/2.1',
+        // might be true if editing inline
+        stripHTMLEnvelope: editor.elementMode === CKEDITOR.ELEMENT_MODE_INLINE,
+        fromHTML: false,
+        toHTML: true,
+        text: content
+      }
+    })
+      .done(function (content) {
+        let dd = new diffDOM.DiffDOM();
+        let newContent;
+        if (editor.elementMode === CKEDITOR.ELEMENT_MODE_INLINE) {
+          let tmpServerBody = $('&lt;body&gt;&lt;/body&gt;').append(content);
+          let tmpCurrentBody = $('&lt;body&gt;&lt;/body&gt;').append(currentHTMLContent);
+          let diff = dd.diff(diffDOM.nodeToObj(tmpCurrentBody[0]), diffDOM.nodeToObj(tmpServerBody[0]))
+          dd.apply(tmpCurrentBody[0], diff);
+          newContent = tmpCurrentBody[0].innerHTML;
+        } else {
+          let currentDOM = $(new DOMParser().parseFromString(currentHTMLContent, 'text/html'));
+          let serverDOM = $(new DOMParser().parseFromString(content, 'text/html'));
+          // Compare the content of the body and apply the diff. We want to preserve the head.
+          let diffAgainst = serverDOM.find('body')[0];
+          let diffFrom = currentDOM.find('body')[0];
+          let diff = dd.diff(diffDOM.nodeToObj(diffFrom), diffDOM.nodeToObj(diffAgainst))
+          dd.apply(diffFrom, diff);
+          newContent = new XMLSerializer().serializeToString(currentDOM[0]);
+        }
+        let cursorRanges = editor.getSelection().getRanges();
+        editor.setData(newContent);
+        editor.focus();
+        editor.getSelection(true).selectRanges(cursorRanges);
+
+      })
+  };
+  let handleCKESource = function (content, editor) {
+    let cursorPosition = $(editor.element.$).prop('selectionStart');
+    console.log(cursorPosition);
+    let editorData = editor.getData();
+    if (content == editorData) {
+      return;
+    }
+    editor.setData(content);
+    $(editor.element.$).focus();
+    $(editor.element.$).prop('selectionEnd', cursorPosition);
+  };
+  let handleWikiEditor = function (content, editor) {
+    let editorData = editor.val();
+    let cursor = editor.prop('selectionStart');
+    console.log(editor.prop('selectionStart'));
+    if (content == editorData) {
+      return;
+    }
+    editor.val(content);
+    editor.prop('selectionEnd', cursor);
+  };
+  let handleWikiEditorWithHighlight = function (content, editor) {
+    let editorData = editor.getValue();
+    if (content == editorData) {
+      return;
+    }
+    let cursor = editor.getCursor();
```

Comment:
Usually the cursor means a collapsed selection. What if there is a word selected when the user saves? Does ``getCursor()`` give you the selection in that case?